netlogo-core/src/main/lab/StatsExporter.scala

```diff
@@ -0,0 +1,231 @@
+    val metrics = ListBuffer[String]()
+    for (m <- protocol.metrics) {
+      if (numericMetrics(m)) {
+        metrics += (f"[${m}]-mean")
+        metrics += (f"[${m}]-std")
+      }
+    }
+    val headers = protocol.valueSets(0).map(_.variableName) ::: "[step]" :: metrics.toList
+    out.println(headers.map(csv.header).mkString(","))
+    out.flush()
+  }
+
+  def writeTableRow(params: List[Any], stats: List[Any], step: Int) {
+    val writeValues = (params :+ step) ::: stats
+    out.println(writeValues.map(csv.data).mkString(","))
+    out.flush()
+  }
+
+  def process: Unit = {
+    val data = extractData()
+    data match {
+      case Some(d) => {
+        writeExportHeader()
+        writeExperimentHeader()
+
+        for (params <- paramCombinations) {
+          val runData = d(params)
+          val sortedSteps = runData.keys.toList.sorted
+
+          for (step <- sortedSteps) {
+            val values = runData(step)
+            val numMetrics = values(0).length
+            val writeData = ListBuffer[Any]()
+
+            for (i <- 0 until numMetrics) {
+              val metric = protocol.metrics(i)
+
+              if (numericMetrics(metric)) {
+                val metricValues = values.map(_(i)).toList
+                val mean = StatsCalculator.mean(metricValues)
+                writeData += mean
+
+                val std = StatsCalculator.std(metricValues)
+                writeData += {
+                  if (std.isNaN) {
+                    "N/A"
+                  } else {
+                    std
+                  }
+                }
+              }
+            }
+            writeTableRow(params, writeData.toList, step)
+          }
+        }
+      }
+      case None =>
+    }
+  }
+
+  def extractData(): Option[Data] = {
+    if (tableExporter != null) {
+      Some(extractFromTable(exporterFileNames(tableExporter)))
+    } else if (spreadsheetExporter != null) {
+      Some(extractFromSpreadsheet(exporterFileNames(spreadsheetExporter)))
+    } else {
+      Some(extractFromSpreadsheet(spreadsheetExporter))
+    }
+  }
+
+  private def handleNonNumeric(entry: String, metricIndex: Int): Double = {
+    try {
+      entry.toDouble
+    } catch {
+      case _: java.lang.NumberFormatException => {
+        if (metricIndex >= 0) {
+          numericMetrics(protocol.metrics(metricIndex)) = false
+        }
+        Double.NaN
+      }
+    }
+  }
+
+  private def extractFromTable(fileName: String): Data = {
+    val bufferedReader: BufferedReader = new BufferedReader(new FileReader(fileName))
+    var line = ""
+    var row = 0
+    val data = new HashMap[List[Any], DataPerStep]() // use a hashmap because we don't know how many runs there are
+    while ({line = bufferedReader.readLine; line} != null) {
+      val params = ListBuffer[Any]()
+      val measurements = ListBuffer[Double]()
+      // The first 6 contain file data, and the 7th row is the header names
+      if (row > 6) {
```

Comment:
I think that a constant such as `HEADER_LENGTH` should be used instead of 6 for any code that reference this number. That way if we change the HEADER_LENGTH we only need to do it in one place. We should do the same thing for any other "special numbers" we run into.
see `./netlogo-core/src/main/fileformat/LabLoader.scala` for an example of setting a constant that can be shared.